src/test/java/io/kestra/plugin/serdes/avro/infer/InferAvroSchemaFromIonTest.java

```diff
@@ -0,0 +1,176 @@
+package io.kestra.plugin.serdes.avro.infer;
+
+import org.junit.jupiter.api.Test;
+import org.skyscreamer.jsonassert.JSONAssert;
+import org.skyscreamer.jsonassert.JSONCompareMode;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+public class InferAvroSchemaFromIonTest {
+    @Test
+    void simple_string_should_be_optional() throws IOException {
+        this.run(
+            """
+                {myString: "hello"}
+                """,
+            """
+                {"fields": [{"name": "myString", "type": ["null","string"]}]}
+                """
+        );
+    }
+
+    @Test
+    void array_of_strings() throws IOException {
+        this.run(
+            """
+                ["one", "two", "three"]
+                """,
+            """
+                {"type": "array", "items": [ "null", "string" ]}
+                """);
+    }
+
+    @Test
+    void complex_nested_records_with_clashing_field_names_do_not_raise_error() throws IOException {
+        this.run(
+            """
+                {
+                    myField: "hey",
+                    firstObject: {
+                        myField: {
+                            myNestedField: "heya"
+                        },
+                        myNestedObject: {
+                            myField: "heyo"
+                        }
+                    },
+                    secondObject: {
+                        myField: "heyu"
+                    }
+                }
+                """,
+            """
+                { }
+                """
+        );
+    }
+
+    @Test
+    void array_of_objects() throws IOException {
+        this.run(
+            """
+                {
+                    myArray: [
+                        {myName: "one"},
+                        {myName: "two"}
+                    ]
+                }
+                """,
+            """
+                {
+                   "fields": [
+                     {
+                       "name": "myArray",
```

Comment:
Missing default value `null` make sure that all fields have a null default value

Reply:
which one ? I don't see it

Reply:
For example, when you have an Array in your tests you have: 

Input
```json
 ["one", "two", "three"]
```
Output

```json
 {"type": "array", "items": [ "null", "string" ]}
```
But, I think the output should

```json
 ["null", {"type": "array", "items": [ "null", "string" ]} ]
```